pkg/solana/fees/block_history_test.go

```diff
@@ -15,82 +16,442 @@ import (
 
 	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/mathutil"
 	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
 
 	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
 	clientmock "github.com/smartcontractkit/chainlink-solana/pkg/solana/client/mocks"
 	cfgmock "github.com/smartcontractkit/chainlink-solana/pkg/solana/config/mocks"
 )
 
-func TestBlockHistoryEstimator(t *testing.T) {
+func TestBlockHistoryEstimator_InvalidBlockHistorySize(t *testing.T) {
+	// Helper variables for tests
 	min := uint64(10)
 	max := uint64(1000)
+	defaultPrice := uint64(100)
+	invalidDepth := uint64(0) // Invalid value to trigger default case
+	pollPeriod := 1 * time.Second
 
+	// Setup
 	rw := clientmock.NewReaderWriter(t)
 	rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
 		return rw, nil
 	})
 	cfg := cfgmock.NewConfig(t)
-	cfg.On("ComputeUnitPriceDefault").Return(uint64(100))
-	cfg.On("ComputeUnitPriceMin").Return(min)
-	cfg.On("ComputeUnitPriceMax").Return(max)
-	cfg.On("BlockHistoryPollPeriod").Return(100 * time.Millisecond)
-	lgr, logs := logger.TestObserved(t, zapcore.DebugLevel)
+	setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, invalidDepth)
+	lgr, _ := logger.TestObserved(t, zapcore.DebugLevel)
 	ctx := tests.Context(t)
+	estimator := initializeEstimator(t, rwLoader, cfg, lgr, ctx)
```

Comment:
I think since you've removed the eventually's I don't think we need an observed logger anymore. Think you can just do something like this.
```suggestion
	ctx := tests.Context(t)
	estimator := initializeEstimator(t, rwLoader, cfg, logger.Test(t), ctx)
```